python/cugraph/cugraph/gnn/dgl_extensions/utils/sampling.py

```diff
@@ -0,0 +1,181 @@
+    if is_mg:
+        # FIXME: Can not switch to renumber = False
+        # For MNMG Algos
+        # Remove when https://github.com/rapidsai/cugraph/issues/2437
+        # lands
+        create_subgraph_f = subgraph.from_dask_cudf_edgelist
+        renumber = True
+        edge_list = edge_list.persist()
+        src_range = edge_list[src_n].min().compute(), edge_list[src_n].max().compute()
+
+    else:
+        # Note: We have to keep renumber = False
+        # to handle cases when the seed_nodes is not present in subgraph
+        create_subgraph_f = subgraph.from_cudf_edgelist
+        renumber = False
+        src_range = edge_list[src_n].min(), edge_list[src_n].max()
+
+    create_subgraph_f(
+        edge_list,
+        source=src_n,
+        destination=dst_n,
+        edge_attr=eid_n,
+        renumber=renumber,
+        # FIXME: renumber=False is not supported for MNMG algos
+        legacy_renum_only=True,
+    )
+
+    return subgraph, src_range
+
+
+def sample_multiple_sgs(
+    sgs,
+    sample_f,
+    start_list_d,
+    start_list_dtype,
+    edge_dir,
+    fanout,
+    with_replacement,
+):
+    start_list_types = list(start_list_d.keys())
+    output_dfs = []
+    for can_etype, (sg, start_list_range) in sgs.items():
+        can_etype = _convert_can_etype_s_to_tup(can_etype)
+        if _edge_types_contains_canonical_etype(can_etype, start_list_types, edge_dir):
+            if edge_dir == "in":
+                subset_type = can_etype[2]
+            else:
+                subset_type = can_etype[0]
+            output = sample_single_sg(
+                sg,
+                sample_f,
+                start_list_d[subset_type],
+                start_list_dtype,
+                start_list_range,
+                fanout,
+                with_replacement,
+            )
+            output_dfs.append(output)
+    if len(output_dfs) == 0:
+        empty_df = cudf.DataFrame({"sources": [], "destinations": [], "indices": []})
+        return empty_df.astype(cp.int32)
+
+    if isinstance(output_dfs[0], dask_cudf.DataFrame):
+        return dask_cudf.concat(output_dfs, ignore_index=True)
+    else:
+        return cudf.concat(output_dfs, ignore_index=True)
+
+
+def sample_single_sg(
+    sg,
+    sample_f,
+    start_list,
+    start_list_dtype,
+    start_list_range,
+    fanout,
+    with_replacement,
+):
+    if isinstance(start_list, dict):
+        start_list = cudf.concat(list(start_list.values()))
+
+    # Uniform sampling fails when the dtype
+    # of the seed dtype is not same as the node dtype
+    start_list = start_list.astype(start_list_dtype)
+
+    # Filter start list by ranges
+    # to enure the seed is with in index values
+    # see below:
+    # https://github.com/rapidsai/cugraph/blob/branch-22.12/cpp/src/prims/per_v_random_select_transform_outgoing_e.cuh
+    start_list = start_list[
+        (start_list >= start_list_range[0]) & (start_list <= start_list_range[1])
+    ]
```

Comment:
This is the logic change in this PR to fix :  https://github.com/rapidsai/cugraph/issues/2828 . Also see test added in this PR